alpha/declcfg/load.go

```diff
@@ -111,20 +115,113 @@ func walkFiles(root fs.FS, fn func(root fs.FS, path string, err error) error) er
 // If LoadFS encounters an error loading or parsing any file, the error will be
 // immediately returned.
 func LoadFS(root fs.FS) (*DeclarativeConfig, error) {
-	cfg := &DeclarativeConfig{}
-	if err := WalkFS(root, func(path string, fcfg *DeclarativeConfig, err error) error {
+	if root == nil {
+		return nil, fmt.Errorf("no declarative config filesystem provided")
+	}
+
+	concurrency := runtime.NumCPU()
+
+	var (
+		fcfg     = &DeclarativeConfig{}
+		pathChan = make(chan string, concurrency)
+		cfgChan  = make(chan *DeclarativeConfig, concurrency)
+	)
+
+	// Create an errgroup to manage goroutines. The context is closed when any
+	// goroutine returns an error. Goroutines should check the context
+	// to see if they should return early (in the case of another goroutine
+	// returning an error).
+	eg, ctx := errgroup.WithContext(context.Background())
```

Comment:
Yeah, I considered that. This function never had anything in its call stack that had a context parameter, so it was never plumbed through. Perhaps it's worth the breaking API change to go ahead and do this (this package is in an `alpha` subtree). If we were concerned about that, we could deprecate this function and add a new `LoadFSContext` function or something.

I think I'd vote for just changing `LoadFS` to accept a context.

Reply:
I attempted this in another commit and realized two things:
1. The change cascades all over the place
2. It may make sense to add context to other walk/load functions in this package.

Given that, I'd like to handle passing context in a separate PR.